app/javascript/shared/components/tags.jsx

```diff
@@ -30,7 +30,9 @@ class Tags extends Component {
   constructor(props) {
     super(props);
 
-    this.debouncedTagSearch = debounce(this.handleInput.bind(this), 150);
+    this.debouncedTagSearch = debounce(this.handleInput.bind(this), 150, {
```

Comment:
can we abstract this stuff into some re-usable constants now? 😬

Reply:
The 150? That's not currently used anywhere else. The other two places I know that we use debounce it is 300.

Reply:
Yeah, but we are now starting to add `leading: true` to a few places and touching the debounce code again, so, since we're in the code already, I think we should clean it up a bit.

Could make a shared global for the option that you added: `const defaultDebounceConfig = { leading: true }`, which can be used in .

And a `DEFAULT_DEBOUNCE_TIME = 300`, which you can explicitly override for the `debcouncedTagSearch` method.

I know it's a little annoying to do, but now that we're explicitly using `debounce` in three different JS files, I can see this being super helpful down the road, and it'll be much clearer and consistent.

Reply:
Ok, @mstruve and I chatted about this, and while I do think we should clean up our usage of `debounce` across our components, this might end up being more work than it's worth just to merge this PR at this moment. 

I've added this clean up task to my own personal "cool down period" todo list. I'll likely just create a util function for this so that everyone else going forward can use that, instead of having to figure out how to implement `debounce` with their own individual arguments.

Reply:
Agreed.

+1 on adding an utility function that decorates the needed function and options with a default so that we import `debounce` in only one place and developers don't need to go check the parameters on https://lodash.com/docs#debounce each time!

Reply:
@rhymes great, thanks for the affirmation! 😅 i'll work on this next week during the ❄️COOL DOWN ❄️